File: server/dotnet/sdk-dotnet-example/Business/DTOs/CreatePayments/CreatePaymentRequestDto.cs
using Business.Domain.Common;
using Business.Domain.Common.Enums;
using Business.Domain.Enums.Payments;
using Business.Domain.Payments;

namespace Business.DTOs.CreatePayments;

public class CreatePaymentRequestDto
{
    public decimal? Amount { get; init; }
    
    public Currency? Currency { get; init; }
    
    public PaymentMethodType? Method { get; init; }

    public string? HostedTokenizationId { get; init; }
    
    public int? PaymentProductId { get; set; }
    
    public AddressDto? ShippingAddress { get; init; }  = new();
    
    public AddressDto? BillingAddress { get; init; }  = new();

    public Card? Card { get; init; }  = new();

    public Mandate? Mandate { get; set; }  = new();
}

File: server/nodejs/src/business/dtos/payment/create-payment/create-payment-request-dto.ts
import { Currency } from '../../../domain/common/enums/currency';
import { PaymentMethodType } from '../../../domain/enums/payments/payment-method-type';
import { AddressDto } from '../../common/address-dto';
import { Card } from '../../../domain/payments/card';
import { Mandate } from '../../../domain/payments/mandate';

export type CreatePaymentRequestDto = {
    amount?: number | null;
    currency?: Currency | null;
    method?: PaymentMethodType | null;
    hostedTokenizationId?: string | null;
    shippingAddress?: AddressDto | null;
    billingAddress?: AddressDto | null;
    card?: Card | null;
    mandate?: Mandate | null;
    paymentProductId?: number | null;
};
